core/src/main/java/hudson/tools/DownloadFromUrlInstaller.java

```diff
@@ -148,6 +146,16 @@ public String getId() {
             if(d==null)     return Collections.emptyList();
             return Arrays.asList(((InstallableList)JSONObject.toBean(d,InstallableList.class)).list);
         }
+        
+        /**
+         * The UI uses this information to populate the drop-down.
+         * 
+         * @return JSON array of installable tools, never null
+         */
+        public JSONArray getInstallablesJson() throws IOException {
+            JSONObject d = Downloadable.get(getId()).getData();
```

Comment:
<code>Downloadable.get()</code> may return null.
It is a mistake in the original code, but it makes sense to fix it.